udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -85,10 +90,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   /**
-   * Obtains parameter from Comments typing field.
+   * Gets count for comments to be shown. Only numbers 1 to 50 can be an input.
+   * Returns null on error.
    */
+  private Integer getCommentCount(HttpServletRequest request) {
+    // Get the input from the form.
+    String commentCountString = request.getParameter("comment-count");
+
+    // Convert the input to an int.
+    Integer commentCount;
+    try {
+      commentCount = Integer.parseInt(commentCountString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + commentCountString);
+      return null;
+    }
 
-  private String getRequestParameter(HttpServletRequest request, String comment, String defaultValue) {
+    return commentCount;
```

Comment:
I see that the comment count is limited to >0 in other parts of the code, but functions are typically not expected to be aware of constraints made in other parts of a codebase (it would be too complex to keep track), and functions should be self-contained. For example, this function could in the future be called through a different code path, or the limits set in other parts of code may change.

So when defining API it helps to consider the full range of inputs; in this case, what you'd like to do if commentCount is 0 or negative.

Reply:
Done

Reply:
Set to return 1 comment.